src/ensembl/src/content/app/browser/browser-tabs/BrowserTabs.tsx

```diff
@@ -66,7 +66,7 @@ const BrowserTabs: FunctionComponent<BrowserTabsProps> = (
     });
 
     setClickHandlers(callbacks);
-  }, [props.drawerOpened, props.genomeSelectorActive]);
+  }, [props.drawerOpened, props.genomeSelectorActive, props.ensObjectInfo]);
```

Comment:
Why do we want clickHandlers to be inside the component state updated with useEffect? Why create clickHandlers directly inside the body of the component function?

Reply:
I'm not sure why it is done this way but I think this function could just be put inside the body of the component function.

Also, I think it is better to do the refactoring on a separate PR as it could slow down the hot fix we are trying to make and also introduce new bugs.

Reply:
```
  const handleTabClick = (value: TrackType) => {
    if (props.genomeSelectorActive || !props.ensObjectInfo.genome_id) {
      return;
    }

    if (props.drawerOpened === true) {
      props.toggleDrawer(false);
    }

    props.selectBrowserTabAndSave(value);
  }
```

and in the `dd` element:

```
        <dd
          className={getBrowserTabClasses(value)}
          key={value}
          onClick={() => handleTabClick(value)}
        >
```

no need for an object of click handlers. Isn't going to slow anything down. And it's just a translation of what's written inside useEffect into a slightly more recognisable form.

Reply:
That's done.

Reply:
🎉